### DISCUSSION TITLE
Cascade Swin T , Object 개수 outlier 제거 학습 결과

### DISCUSSION THREAD

**Discussion by JiyouSeo** [boostcampaitech2/object-detection-level2-cv-06#130] [Show and tell]
#113 에서 계획했던 학습의 결과를 공유드립니다.

1. Original train set으로 학습 :  cascade_rcnn_swin_base_22kpretrained 
2. Object 15개이하의 train set으로 학습 : cascade_rcnn_swin_base_22kpretrained_lessequal15
3. Object 27개이하의 train set으로 학습 : cascade_swin_base_22kpretrained_lessequal27
4. Object 40개이하의 train set으로 학습 : cascade_swin_base_22kpretrained_lessequal40

**mAP_s**
Small에서는 Outlier 제거를 적게할 수록 mAP_s가 높게 나오게 되었습니다. 이는 Object 개수가 많을 수록 Small에 해당하는 Object가 많지 않았을까 하는 생각입니다.
![image](https://user-images.githubusercontent.com/61641072/136952290-5ba0ae97-5c10-499f-aaf9-8f54614ae329.png)

**mAP_m**
Medium에서는 Object의 개수가 27개 이하인 train set을 학습하였을 때가 가장 높게 수렴하였습니다.
15개 이하의 train set을 제외하고는, Object의 개수를 더 적게 학습할 수록 Medium이 높게 수렴함을 알 수 있었습니다.
![image](https://user-images.githubusercontent.com/61641072/136952327-e2e2f8f4-0ad0-4b4b-81c6-296cf6687eca.png)

**mAP_l**
Large에서는 이렇다할 규칙을 찾지 못하였습니다. Outlier를 제거하지 않은 train set으로 학습한 것이 가장 높게 나오게 되었습니다..
![image](https://user-images.githubusercontent.com/61641072/136952357-67209135-65a0-4413-b9d4-50011e360be1.png)

mAP_50은 Large가 Small, Medium보다 절대적 수치가 높기 떄문에, Large에서 가장 높은 mAP를 보여주었던 
Original train set으로 학습한 모델이 가장 높게 나왔습니다.
![image](https://user-images.githubusercontent.com/61641072/136952381-ca005cf4-ec86-4c95-bbac-d4e645a11db6.png)

**결론**
Outlier를 제거하면 성능이 올라갈 것 이라는 #113에서의 기대와는 달리, 오히려 성능의 하락을 가져오게 되었습니다.
Object 개수만으로 접근하기보다,  mAP_s,m,l의 기준인 Area를 기준으로 train set을 나누어 진행할 계획입니다.

**Comment by jaehwan-AI** [comment]
1. mAP_s 수치를 보면 최대 0.1도 안되는 적은 수치들의 비교이기 때문에 값이 증가했다 하더라도 전체적인 mAP에는 영향을 미치지 않을 것 같다고 생각합니다.
2. mAP_m : `15개 이하의 train set을 제외하고는, Object의 개수를 더 적게 학습할 수록 Medium이 높게 수렴함을 알 수 있었습니다.` 라면 16개의 train set에서 가장 mAP_m이 높게 수렴한다는 말씀이실까요?
3. bbox의 개수와 bbox area와의 상관관계를 시각화 해보면 mAP_s, mAP_m, mAP_l과 bbox 개수와의 관계를 더 살펴볼 수 있을 것 같습니다.
4. outlier라고 생각했지만 제거한 데이터에 학습에 도움이 되는 요소가 담겨있을 수도 있습니다. 지웠던 데이터들을 살펴보는 작업도 해당 결과를 분석하는 과정에서 필요할 것 같습니다.

**Reply by JiyouSeo** [reply]
1. 네, mAP_s의 수치를 올리는 것보다 m,l의 수치를 올리는 것이 유리할 것 같습니다.
2. mAP_m 그래프를 보시면 15개 이하의 train set은 이유를 알 수 없지만, mAP_m이 다른 세 가지 경우보다 낮은 것을 보실 수 있습니다. 따라서 제외하고 생각해보았을 때, 27이하, 40이하, 오리지널 순으로 mAP_m이 높게 수렴하는 기조에 포커스를 두었습니다.
3. 좋은 조언입니다 ! 내일 한번 시각화 해보아야겠습니다.
4. 사실 그 부분이 조금 마음에 걸렸습니다. 하지만 저 그래프 상으로는 제 생각에 object 개수만을 조정해서 mAP를 올리는데는 한계가 있을 것 같습니다. 
   40개 이상의 outlier들을 시각화했던 부분은 #113 에 있긴한데, 15, 27이하도 한번씩 찍어보겠습니다.

**Reply by jaehwan-AI** [reply]
lessequal15, 27, 40 순으로 데이터 수가 증가하는 것이죠?

**Reply by JiyouSeo** [reply]
네 맞습니다!
